salesforce/backend/enterprise.py

```diff
@@ -0,0 +1,46 @@
+import base64
+from hashlib import sha256
+from typing import Optional
+from django.conf import settings
+
+
+class LicenseError(Exception):
+    pass
+
+
+def is_enterprise_license(msg: Optional[str] = None, required: int = 1, key: str = '') -> None:
+    """Check that a valid license key for the enterprise version exists"""
+    key = key or getattr(settings, 'DJSF_LICENSE_KEY', '//')
+    text, level_text, signature_text = key.rsplit('/', 2)
+    level = int(level_text or 1) if text else 0
+    if level < max(required, 1):
+        raise PermissionError(msg or "This command requires the enterprise license key")
+
+    # This check is an integer arithmetic puzzle that does not use anything except Python
+    # and Django settings. It is evaluated in less than half milisecond on a notebook CPU.
+    # There is no cryptography, no network access. The license key never expires.
+    # It is a safe code. Please do not discuss publicly about other aspects how the
+    # license key is weak or strong. Many easier ways exist how to skip a verification
+    # without analyzing this. :-)
+
+    # A legal way to skip the enterprise check is to use django-salesforce-agpl and
+    # accept the restrictive AGPL licence that requires you provide all your Django
```

Comment:
Note that the license.txt in the repo root still uses the MIT license.

Reply:
Seems like forking and removing the protection wouldn't be allowed by GPL either. I don't know if I understand what the goal is of the license change.

My attitude is "don't break into my house because I asked you not to, not because there are bars on the windows"

Reply:
Exactly. Open doors cannot be kicked out. That is why I initially thought about name "sponsor_motivation()", not "...key". I think a some short permissive license is fine with a short restriction related to the enterprise-key code that should be not removed or evaded because it is important for the project.
The AGPL license however (as a copyleft license) also doesn't protect the author's work from copying, modifying, sharing. It establishes a non-dispensable mandatory freedom that everybody who uses that code must share with others and publish also all his/her code under the same copyleft license and give other users the same right on his work, even to end users who do not need a software and only use a service by the network. "Network use is distribution" is a principle of AGPL. Copyleft makes a strong border between "free" open-source world and commercial world. People should not remove the enterprise-key code, but they can use, modify, share... a version under AGPL where the enterprise check is removed by us.